From 1febf1504ddaf7fe4bcf6e79227190e005e1d63b Mon Sep 17 00:00:00 2001
From: pdiopX <piotrx.diop@intel.com>
Date: Mon, 27 May 2013 10:07:01 +0200
Subject: Add configuration facilities for mute strategy for shutter sound

BZ: 108930
[CLEAN]

Allow to enable/disable mute strategy for ENFORCED_AUDIBLE streams (camera
shutter sound).

Category: device enablement
Domain: AUDIO_MEDIA-Audio Flinger
Origin: internal
Upstream-Candidate: yes
Change-Id: I5d373d4ac78df5186037247293575bdee706eda6
Orig-Change-Id: I45cb645e9f53748aafaa81631df1ed805eff1983
Signed-off-by: pdiopX <piotrx.diop@intel.com>
---
 services/audioflinger/AudioPolicyService.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/services/audioflinger/AudioPolicyService.cpp b/services/audioflinger/AudioPolicyService.cpp
index 35e816b..9fc4309 100644
--- a/services/audioflinger/AudioPolicyService.cpp
+++ b/services/audioflinger/AudioPolicyService.cpp
@@ -96,6 +96,10 @@ AudioPolicyService::AudioPolicyService()
     if (rc)
         return;
 
+    property_get("ro.camera.sound.forced", value, "0");
+    forced_val = strtol(value, NULL, 0);
+    mpAudioPolicy->set_can_mute_enforced_audible(mpAudioPolicy, !forced_val);
+
     ALOGI("Loaded audio policy from %s (%s)", module->name, module->id);
 
     // load audio pre processing modules
-- 
1.7.4.1

